packages/react/src/components/DataTable/TableRow.tsx

```diff
@@ -31,7 +31,14 @@ const TableRow = (props: TableRowProps) => {
     [`${prefix}--data-table--selected`]: props.isSelected,
   });
   const cleanProps = {
-    ...omit(props, ['ariaLabel', 'onExpand', 'isExpanded', 'isSelected']),
+    ...omit(props, [
+      'ariaLabel',
+      'aria-Label',
```

Comment:
I think we're fine regardless, but for consistency 

```suggestion
      'aria-label',
```